### PR TITLE
Add fix to handle rule template instance counts properly

### PR DESCRIPTION
**Pull request by minudika**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 1]
```diff
@@ -1277,31 +1289,41 @@ private void updateArtifactCount(String businessRuleUUID, int artifactCount)
...
+                (deploymentStatus != TemplateManagerConstants.PARTIALLY_DEPLOYED)) {
+            if (businessRule instanceof BusinessRuleFromTemplate) {
+                BusinessRuleFromTemplate businessRuleFromTemplate = (BusinessRuleFromTemplate) businessRule;
+                String instanceLimit = getRuleTemplate(businessRuleFromTemplate).getInstanceCount();
+                count = getInstanceCount(businessRuleFromTemplate.getRuleTemplateUUID());
                 if ("one".equalsIgnoreCase(instanceLimit) && count >= 1) {
                     throw new TemplateInstanceCountViolationException("Rule Template '" +
-                            (ruleTemplate.getUuid() + "' can be instantiated " +
-                                    "only once and have been already instantiated."));
+                            ((BusinessRuleFromTemplate) businessRule).getRuleTemplateUUID() + "' can be instantiated " +
```
Comment (tishan89) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: can't we use already casted instance

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 2]
```diff
@@ -1277,31 +1289,41 @@ private void updateArtifactCount(String businessRuleUUID, int artifactCount)
...
+                            "only once and have been already instantiated.");
+                }
+            } else if (businessRule instanceof BusinessRuleFromScratch) {
+                BusinessRuleFromScratch businessRuleFromScratch = (BusinessRuleFromScratch) businessRule;
+                List<RuleTemplate> ruleTemplates =
+                        (List<RuleTemplate>) getInputOutputRuleTemplates(businessRuleFromScratch);
```
Comment (tishan89) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Can we change this method to return a List?

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java (hunk 1 above)
Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Fixed with PR #1205

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java (hunk 2 above)
Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Fixed with PR #1205

**Pull request merged**